set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -104,11 +104,14 @@
 		webUserSet.setVisibility(VisibilityTypes.PRIVATE.getName());
 	    }
 
+	    getUserSetService().validateFavoriteUserSet(webUserSet, null);
```

Comment:
this call should be integrated within the validateWebUserSet method

Reply:
Then we would need to add additional parameter to validateWebUserSet method. Also, I believe, new method also better separates logical parts of validation.

Reply:
Why would you need a different parameter? the type is included in the WebUserSet, that is the property used to verify if the set is a boomark folder

Reply:
When we use this method for update method, and in this case some required fields in webUserSet is null and we need to pass user set existing in database to get these fields. E.g. in update method creator is empty in webUserSet and we need creator for update use case, because it is not allowed to change the type. If type is present in the request body, it must match the value from the object stored in the database. For this reason, I believe, we need the second parameter for common method.

Reply:
Update method is not creating a new set. . For the update method we retrieve first the set from the database based on its ID. 
The type and the creator are already available in set retrieved from the database. 
Only if the type in the request body is BookmarksFolder and it is different from the Type in database object, only in this case we have to verify if a BookmarkFolder exist in the database for the same user (creator is to be taken from the existingUserSet)